node/pkg/fetcher/accumulator.go

```diff
@@ -21,16 +23,33 @@ func (a *Accumulator) Run(ctx context.Context) {
 	a.cancel = cancel
 	a.isRunning = true
 
+	// dummy channel to signal accumulator workers to batch insert data to db
+	jobChannel := make(chan struct{}, 100)
+	for i := 0; i < ACCUMULATOR_WORKER_COUNT; i++ {
+		go a.accumulatorWorker(accumulatorCtx, jobChannel)
+	}
+
 	ticker := time.NewTicker(DefaultLocalAggregateInterval)
 	defer ticker.Stop()
 
 	for {
 		select {
+		case <-ticker.C:
+			jobChannel <- struct{}{}
 		case <-ctx.Done():
 			log.Debug().Str("Player", "Fetcher").Msg("fetcher local aggregates channel goroutine stopped")
 			return
-		case <-ticker.C:
-			go a.accumulatorJob(ctx)
+		}
+	}
+}
+
+func (a *Accumulator) accumulatorWorker(ctx context.Context, jobChannel <-chan struct{}) {
+	for {
+		select {
+		case <-jobChannel:
+			a.accumulatorJob(ctx)
+		case <-ctx.Done():
+			return
```

Comment:
wouldn't this be signalling all 3 accumulatorWorkers with the same jobs? please tell me if I'm missing anything

Reply:
maybe using mutex would make it safer 🤔

Reply:
ah I wondered the same initially but apparently only one of the workers will get triggered because channels in Go provide a synchronization mechanism that ensures that each message sent through the channel is received by exactly one receiver. 

So, each time a message is sent to `jobChannel`, only one of the waiting goroutines will receive that message while the other goroutines will continue waiting for the next message to be sent to the channel.

Reply:
can you include some test code so that we can make sure single job can be processed at the same time?

Reply:
you could add some checks if there aren't any duplicate data inserted